@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -380,30 +392,111 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
+                            segmentOffsets.push(lastSegmentOffset);
+                        }
+
+                        tmpV3.sub(this.m_decodeInfo.center);
+                        localLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
+                        tmpV4.sub(this.m_decodeInfo.center);
+                        localLine.push(tmpV4.x, tmpV4.y, tmpV4.z);
+
+                        localLineSegments!.push(localLine);
+                        worldLines.push(worldLine);
+                        uvs.push(lineUvs);
+                        offsets.push(segmentOffsets);
+                    }
                 }
-                if (hasUntiledLines) {
-                    // Find where in the [0...1] range relative to the line our current vertex lies.
-                    const offset = this.findRelativePositionInLine(tmpV3, untiledLine) / lineDist;
-                    lineOffsets.push(offset);
+
+                if (!hasUntiledLines && lineLength > 0) {
+                    // Scale down each individual segment to range [0..1] for whole line.
+                    for (const segOffsets of offsets) {
+                        segOffsets.forEach((offs, index) => {
+                            segOffsets[index] = offs / lineLength;
+                        });
+                    }
                 }
-                tmpV3.sub(this.m_decodeInfo.center);
-                localLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
-            });
-            localLines.push(localLine);
-            worldLines.push(worldLine);
-            uvs.push(lineUvs);
-            offsets.push(lineOffsets);
+            }
+
+            // Add continuous lines
+            if (hasContinuousLineSegments) {
+                const localLine: number[] = [];
+                const worldLine: number[] = [];
+                const lineUvs: number[] = [];
+                const lineOffsets: number[] = [];
+                polyline.positions.forEach(pos => {
+                    webMercatorTile2TargetWorld(extents, this.m_decodeInfo, pos, tmpV3);
+                    worldLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
+
+                    if (computeTexCoords) {
+                        const { u, v } = computeTexCoords(pos, extents);
+                        lineUvs.push(u, v);
+                    }
+                    if (hasUntiledLines) {
+                        // Find where in the [0...1] range relative to the line our current vertex
+                        // lines.
+                        const offset =
+                            this.findRelativePositionInLine(tmpV3, untiledLine) / lineDist;
+                        lineOffsets.push(offset);
+                    }
+                    tmpV3.sub(this.m_decodeInfo.center);
+                    localLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
+                });
+                localLines.push(localLine);
+                worldLines.push(worldLine);
+                uvs.push(lineUvs);
```

Comment:
Maybe check before if this is worth to collect uvs...
`this.getTextureCoordinateType(technique)`

Here we create array of empty arrays...

Reply:
That is old code, if I am not mistaken. I am not going to touch this.